### ISSUE TITLE
Adjust collision shapes and onColShapeHit dimension logic

### ISSUE THREAD

**Issue by Woovie** [multitheftauto/mtasa-blue#1209]
I think we could simplify collision shapes by making them dimensionless and instead just return the dimension onColShapeHit. This would break backwards compatibility as the return values for onColShapeHit would change, and collisions would not be able to be moved dimensionally. But, what is the point of moving a collision shape across dimensions if it applies to all anyways?

**Comment by qaisjp** [comment]
[setElementDimension](https://wiki.multitheftauto.com/wiki/SetElementDimension) has support for omnipresence:

> You can also use -1 to make the element visible in all dimensions (only valid for objects)

~~If this doesn't work for colshapes right now,~~ we could make it work for colshapes.  That will solve any backwards compatibility problems.

**Comment by qaisjp** [comment]
I forgot that [onColShapeHit](https://wiki.multitheftauto.com/wiki/OnColShapeHit) has a `matchingDimension` parameter.

Col shapes works in all dimensions possibly because of [backwards compat reasons](https://wiki.multitheftauto.com/index.php?title=OnColShapeHit&diff=8100&oldid=6431).

> But, what is the point of moving a collision shape across dimensions if it applies to all anyways?

Assuming this was a historical bug, the `matchingDimension` param is intended to alleviate that.

----

I don't see a reason to adjust onColShapeHit, but if we did, I'd say we should just make `onColShapeHit` only get triggered in the same dimension, add support for `shape.dimension = -1`, and have `matchingDimension` always say `true`.

----
TLDR; generally you can say that col shapes are already dimensionless (in the same manner that water used to ignore dimensions), and the hotfix for that was to add `matchingDimension` as a convenient way to shorten `getElementDimension(source) == getElementDimension(player)`.
